integrated_channels/integrated_channel/exporters/content_metadata.py

```diff
@@ -160,6 +162,88 @@ def _check_matched_content_updated_at(
         return items_to_update
 
+    def _check_matched_content_to_create(
+        self,
+        enterprise_customer_catalog,
+        matched_items
+    ):
+        """
+        Take a list of content keys and create ContentMetadataItemTransmission records. When existed soft-deleted
+        records exist, resurrect them. When created but not-yet-transmitted records exist, include them.
+
+        Args:
+            enterprise_customer_catalog (EnterpriseCustomerCatalog): The enterprise catalog object
+
+            matched_items (list): A list of dicts containing content keys and the last datetime that the respective
+                content was updated
+
+            force_retrieve_all_catalogs (Bool): If set to True, all content under the catalog will be retrieved,
+                regardless of the last updated at time
+        """
+        ContentMetadataItemTransmission = apps.get_model(
+            'integrated_channel',
+            'ContentMetadataItemTransmission'
+        )
+        items_to_create = {}
+        create_transmissions = []
+        for matched_item in matched_items:
+            content_id = matched_item.get('content_key')
+            content_last_changed = matched_item.get('date_updated')
+            past_deleted_transmission = ContentMetadataItemTransmission.objects.filter(
+                enterprise_customer=self.enterprise_configuration.enterprise_customer,
+                plugin_configuration_id=self.enterprise_configuration.id,
+                integrated_channel_code=self.enterprise_configuration.channel_code(),
+                content_id=content_id,
+                deleted_at__isnull=False,
+            ).first()
+            incomplete_transmission = ContentMetadataItemTransmission.objects.filter(
+                enterprise_customer=self.enterprise_configuration.enterprise_customer,
+                plugin_configuration_id=self.enterprise_configuration.id,
+                integrated_channel_code=self.enterprise_configuration.channel_code(),
+                content_id=content_id,
+                created_at__isnull=True,
+            ).first()
+            if past_deleted_transmission:
+                self._log_info(
+                    'Found previously deleted content record while creating record. '
+                    'Marking record as active.',
+                    course_or_course_run_key=content_id
+                )
+                past_deleted_transmission.api_response_status_code = None
+                past_deleted_transmission.api_response_body = None
+                past_deleted_transmission.deleted_at = None
+                past_deleted_transmission.channel_metadata = None
+                past_deleted_transmission.content_last_changed = content_last_changed
+                past_deleted_transmission.enterprise_customer_catalog_uuid = enterprise_customer_catalog.uuid,
+                past_deleted_transmission.save()
```

Comment:
nit: would be nice as a `past_deleted_transmission.resurrect()` convenience function call.